src/pre-offline.ts

```diff
@@ -0,0 +1,11 @@
+import { EsbuildPlugin } from '.';
+import { relative } from 'path';
+export function preOffline(this: EsbuildPlugin) {
+  // Set offline location automatically if not set manually
+  if (!this.serverless?.service?.custom?.['serverless-offline']?.location) {
+    this.serverless.service.custom['serverless-offline'].location = relative(
```

Comment:
@olup I am getting an error here

```bash
 TypeError: Cannot set property 'location' of undefined
      at EsbuildPlugin.preOffline (/Users/benoit/projects/serverless-esbuild/dist/pre-offline.js:9:71)
```
It happens because I do not have the `serverless-offline` key under `custom`.

Adding this solves the issue:

```ts
if (!this.serverless.service.custom['serverless-offline']) {
  this.serverless.service.custom['serverless-offline'] = {};
}
```

Reply:
yes of course
fixing it now